strawberry/extensions/runner.py

```diff
@@ -21,37 +27,59 @@ def __init__(
 
     def _run_on_all_extensions(self, method_name: str, *args, **kwargs):
         for extension in self.extensions:
-            getattr(extension, method_name)(*args, **kwargs)
+            method = getattr(extension, method_name)
```

Comment:
Is there not a way to avoid this `getattr` process?

Reply:
We could let the extension context manager call their `on_` method directly. This results in more code per context manager, but guarantees the method exists:

```python
class RequestContextManager(ExtensionContextManager):
    def __enter__(self):
        for extension in self.extensions:
            extension.on_request_start()

    def __exit__(self, exc_type, exc_val, exc_tb):
        for extension in self.extensions:
            extension.on_request_end()

    async def __aenter__(self):
        async for extension in self.extensions:
            await extension.on_request_end()

    async def __aexit__(self, exc_type, exc_val, exc_tb):
        async for extension in self.extensions:
            await extension.on_request_end()
```

Reply:
Just implemented and pushed this approche. We can revert it if we find a better solution :)